src/panels/KaitoManagePanel.ts

```diff
@@ -0,0 +1,206 @@
+        } finally {
+            this.operating[model] = false;
+        }
+    }
+
+    // Deletes the workspace and redeploys it
+    private async handleRedeployWorkspaceRequest(
+        modelName: string,
+        modelYaml: string,
+        webview: MessageSink<ToWebViewMsgDef>,
+    ) {
+        if (this.operating[modelName]) {
+            vscode.window.showErrorMessage(`Operation in progress for 'workspace-${modelName}'. Please wait.`);
+            return;
+        }
+        try {
+            await this.handleDeleteWorkspaceRequest(modelName, webview);
+            this.operating[modelName] = true;
+            await longRunning(`Re-deploying 'workspace-${modelName}'`, async () => {
+                const tempFilePath = join(tmpdir(), `kaito-deployment-${Date.now()}.yaml`);
+                writeFileSync(tempFilePath, modelYaml, "utf8");
+                const command = `apply -f ${tempFilePath}`;
+                const kubectlresult = await invokeKubectlCommand(this.kubectl, this.kubeConfigFilePath, command);
+                if (failed(kubectlresult)) {
+                    vscode.window.showErrorMessage(`Error deploying 'workspace-${modelName}': ${kubectlresult.error}`);
+                    return;
+                }
+            });
+            vscode.window.showInformationMessage(`'workspace-${modelName}' has been redeployed.`);
+            await this.updateModels(webview);
+        } finally {
+            this.operating[modelName] = false;
+        }
+    }
+
+    // Updates the current state of models on the cluster
+    private async updateModels(webview: MessageSink<ToWebViewMsgDef>) {
+        const command = `get workspace -o json`;
+        const kubectlresult = await invokeKubectlCommand(this.kubectl, this.kubeConfigFilePath, command);
+        if (failed(kubectlresult)) {
+            webview.postMonitorUpdate({
+                clusterName: this.clusterName,
+                models: [],
+            });
+            return;
+        }
+
+        const models = [];
+        const data = JSON.parse(kubectlresult.result.stdout);
+        for (const item of data.items) {
+            const conditions: Array<{ type: string; status: string }> = item.status?.conditions || [];
+            const { resourceReady, inferenceReady, workspaceReady } = getConditions(conditions);
+            models.push({
+                name: item.inference.preset.name,
+                instance: item.resource.instanceType,
+                resourceReady: resourceReady,
+                inferenceReady: inferenceReady,
+                workspaceReady: workspaceReady,
+                age: convertAgeToMinutes(item.metadata?.creationTimestamp),
+            });
+        }
+
+        webview.postMonitorUpdate({
+            clusterName: this.clusterName,
+            models: models,
+        });
+    }
+
+    private async handleMonitorUpdateRequest(models: ModelState[], webview: MessageSink<ToWebViewMsgDef>) {
+        void models;
```

Comment:
Suggestion: does this line has any effect? Please clarify. Are you trying to clear models array?